I do not understand how to get the sonar report from these logs:
```
scan-build: Run 'scan-view /tmp/scan-build-2023-01-10-161330-53449-1' to examine bug reports.
Error: Process completed with exit code 1.
```
 ??

--- a/src/lib/attrs.c
+++ b/src/lib/attrs.c
@@ -201,6 +201,11 @@ bool attr_list_add_int(attr_list *l, CK_ATTRIBUTE_TYPE type, CK_ULONG value) {
     return _attr_list_add(l, type, sizeof(value), (CK_BYTE_PTR)&value, TYPE_BYTE_INT);
 }
 
+bool attr_list_add_int_seq(attr_list *l, CK_ATTRIBUTE_TYPE type, CK_BYTE_PTR value, CK_ULONG len) {
+
+    return _attr_list_add(l, type, len, value, TYPE_BYTE_INT_SEQ);
+}
+
 bool attr_list_add_bool(attr_list *l, CK_ATTRIBUTE_TYPE type, CK_BBOOL value) {
 
     return _attr_list_add(l, type, sizeof(value), &value, TYPE_BYTE_BOOL);
@@ -890,13 +895,13 @@ CK_RV rsa_gen_mechs(attr_list *new_pub_attrs, attr_list *new_priv_attrs) {
     };
 
     if (new_pub_attrs) {
-        bool r = attr_list_add_buf(new_pub_attrs, CKA_ALLOWED_MECHANISMS,
+        bool r = attr_list_add_int_seq(new_pub_attrs, CKA_ALLOWED_MECHANISMS,
                 (CK_BYTE_PTR)&t, sizeof(t));
         goto_error_false(r);
     }
 
     if (new_priv_attrs) {
-        bool r = attr_list_add_buf(new_priv_attrs, CKA_ALLOWED_MECHANISMS,
+        bool r = attr_list_add_int_seq(new_priv_attrs, CKA_ALLOWED_MECHANISMS,
             (CK_BYTE_PTR)&t, sizeof(t));
         goto_error_false(r);
     }
@@ -976,11 +981,11 @@ static CK_RV ecc_gen_mechs(attr_list *new_pub_attrs, attr_list *new_priv_attrs) 
         CKM_ECDSA_SHA512,
     };
 
-    bool r = attr_list_add_buf(new_pub_attrs, CKA_ALLOWED_MECHANISMS,
+    bool r = attr_list_add_int_seq(new_pub_attrs, CKA_ALLOWED_MECHANISMS,
             (CK_BYTE_PTR)&t, sizeof(t));
     goto_error_false(r);
 
-    r = attr_list_add_buf(new_priv_attrs, CKA_ALLOWED_MECHANISMS,
+    r = attr_list_add_int_seq(new_priv_attrs, CKA_ALLOWED_MECHANISMS,
             (CK_BYTE_PTR)&t, sizeof(t));
     goto_error_false(r);
 

--- a/src/lib/attrs.h
+++ b/src/lib/attrs.h
@@ -105,6 +105,21 @@ bool attr_list_add_bool(attr_list *l, CK_ATTRIBUTE_TYPE type, CK_BBOOL value);
 bool attr_list_add_int(attr_list *l, CK_ATTRIBUTE_TYPE type, CK_ULONG value);
 
 /**
+ * Adds a CK_ULONG to the attribute list and adds type data.
+ * @param l
+ *  The list to add to.
+ * @param type
+ *  The attribute type to add.
+ * @param value
+ *  The buffer, can be NULL.
+ * @param len
+ *  The length of the buffer. 0 is treated as NULL value.
+ * @return
+ *  true on success, false otherwise.
+ */
+bool attr_list_add_int_seq(attr_list *l, CK_ATTRIBUTE_TYPE type, CK_BYTE_PTR value, CK_ULONG len);
+
+/**
  * Returns the count items in the attribute list.
  *
  * @param l

--- a/src/lib/db.c
+++ b/src/lib/db.c
@@ -32,6 +32,7 @@
 #include "tpm.h"
 #include "twist.h"
 #include "utils.h"
+#include "typed_memory.h"
 
 #include <openssl/evp.h>
 
@@ -39,7 +40,7 @@
 #define TPM2_PKCS11_STORE_DIR "/etc/tpm2_pkcs11"
 #endif
 
-#define DB_VERSION 7
+#define DB_VERSION 8
 
 #define goto_oom(x, l) if (!x) { LOGE("oom"); goto l; }
 #define goto_error(x, l) if (x) { goto l; }
@@ -2131,6 +2132,69 @@ error:
     return rv;
 }
 
+static CK_RV dbup_handler_from_7_to_8(sqlite3 *updb) {
+
+    /*
+     * Between version 7 and 8 of the DB the following changes need to be made:
+     *
+     * Table tobjects:
+     *
+     * The YAML attributes for a int sequence has to be a yaml sequence.
+     */
+
+
+    CK_RV rv = CKR_GENERAL_ERROR;
+    sqlite3_stmt *stmt = NULL;
+
+    int rc = sqlite3_prepare_v2(updb, "SELECT * from tobjects", -1, &stmt, 0);
+    if (rc != SQLITE_OK) {
+        LOGE("Failed to fetch data: %s", sqlite3_errmsg(updb));
+        goto error;
+    }
+
+    rc = sqlite3_step(stmt);
+    if (rc == SQLITE_DONE) {
+        goto out;
+    } else if (rc != SQLITE_ROW) {
+        LOGE("Failed to step: %s", sqlite3_errmsg(updb));
+        goto error;
+    }
+
+    while (rc == SQLITE_ROW) {
+        tobject *tobj = db_tobject_new(stmt);
+        if (!tobj) {
+            LOGE("Could not process tobjects for upgrade");
+            goto error;
+        }
+
+        /* for each tobject */
+        CK_ATTRIBUTE_PTR a = attr_get_attribute_by_type(tobj->attrs, CKA_ALLOWED_MECHANISMS);
+        CK_BYTE type = type_from_ptr(a->pValue, a->ulValueLen);
+        if (type != TYPE_BYTE_INT_SEQ) {
+            rv = _db_update_tobject_attrs(updb, tobj->id, tobj->attrs);
+        }
+
+        tobject_free(tobj);
+        if (rv != CKR_OK) {
+            goto error;
+        }
+
+        rc = sqlite3_step(stmt);
+        if (rc != SQLITE_ROW && rc != SQLITE_DONE) {
+            LOGE("Failed to fetch data: %s\n", sqlite3_errmsg(updb));
+            goto error;
+        }
+    }
+
+out:
+    rv = CKR_OK;
+
+error:
+    sqlite3_finalize(stmt);
+    return rv;
+}
+
+
 static CK_RV db_backup(sqlite3 *db, const char *dbpath, sqlite3 **updb, char **copypath) {
 
     CK_RV rv = CKR_GENERAL_ERROR;
@@ -2205,7 +2269,8 @@ static CK_RV db_update(sqlite3 **xdb, const char *dbpath, unsigned old_version, 
             dbup_handler_from_3_to_4,
             dbup_handler_from_4_to_5,
             dbup_handler_from_5_to_6,
-            dbup_handler_from_6_to_7
+            dbup_handler_from_6_to_7,
+            dbup_handler_from_7_to_8
     };
 
     /*

